VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs

```diff
@@ -162,5 +166,52 @@ protected virtual void AddParentOrganizations(IndexDocument document, ICollectio
             document.AddFilterableValues("ParentOrganizations", nonEmptyValues);
             document.AddFilterableValue("HasParentOrganizations", nonEmptyValues?.Any() ?? false);
         }
+
+        protected virtual void IndexDynamicProperties(IndexDocument document, ICollection<DynamicObjectProperty> properties)
+        {
+            foreach (var property in properties)
+            {
+                var propertyName = property.Name?.ToLowerInvariant();
+                if (!string.IsNullOrEmpty(propertyName))
+                {
+                    var isCollection = property.IsDictionary || property.IsArray;
+                    IList<object> values = new List<object>();
+
+                    switch (property.ValueType)
```

Comment:
Switch/Case there can be omit there. And also you need to remove redundant calls  `document.Add(new IndexDocumentField)`  with reducing  to one call.
Also need to prevent null reference exception throwing if property,values aren't set.